iceoryx_utils/include/iceoryx_utils/internal/posix_wrapper/access_control.hpp

```diff
@@ -43,8 +43,12 @@ class AccessController
     /// @brief maximum number of permission entries the AccessController can store
     static constexpr int32_t MaxNumOfPermissions = 20;
 
-    /// @brief identifier for a permission entry (user, group, others, ...)
+/// @brief identifier for a permission entry (user, group, others, ...)
+#if defined(QNX) || defined(QNX__) || defined(__QNX__)
```

Comment:
the same logic is applied to `defined(__APPLE__)`.

Reply:
I think for the mac support we need to do some refactoring and introduce something like a platform abstraction where all this types are defined in a single place